src/utils/sources-tree/updateTree.js

```diff
@@ -22,13 +25,55 @@ function newSourcesSet(newSources, prevSources) {
   return uniqSources;
 }
 
+function findFocusedItemInTree(
+  newSourceTree: TreeDirectory,
+  debuggeeHost: ?string,
+  focusedItem: ?TreeNode
+): ?TreeNode {
+  if (focusedItem) {
+    const parts = focusedItem.path.split("/").filter(p => p !== "");
+    let path = "";
+
+    console.log("------ START OF TREE SEARCH ------");
+    focusedItem = parts.reduce((subTree, part, index) => {
+      console.log("SUB TREE:", subTree);
+      console.log("PART:", part);
+      console.log("PARTS:", parts);
+      console.log("IS DIRECTORY:", isPartDir(focusedItem, parts.length, index));
+      path = path ? `${path}/${part}` : part;
+      const { index: childIndex } = findNodeInContents(
+        subTree,
+        createTreeNodeMatcher(
+          part,
+          isPartDir(focusedItem, parts.length, index),
+          debuggeeHost
+        )
+      );
+      console.log("------ END OF LOOP ------");
+      return subTree.contents[childIndex];
+    }, newSourceTree);
+    console.log("------ END OF TREE SEARCH ------");
+    console.log("FOCUSED ITEM:", focusedItem);
+  }
+
+  return focusedItem;
+}
+
+function isPartDir(focusedItem: ?TreeNode, partsLength, index) {
+  if (focusedItem && isDirectory(focusedItem)) {
+    return true;
+  }
+  return partsLength - 1 != index;
```

Comment:
I'll probably leverage the actual [partIsFile helper](https://github.com/devtools-html/debugger.html/blob/master/src/utils/sources-tree/utils.js#L75) which looks like it's doing a similar thing. However, `url` will only be available if `focusedItem` is a source _I think_.